File: ArtKotlinCoreV2/src/main/java/net/dev/art/kotlin/core/artkotlincore/ArtPlugin.java
package net.dev.art.kotlin.core.artkotlincore;

import org.bukkit.plugin.java.JavaPlugin;

public abstract class ArtPlugin extends JavaPlugin {

    private static ArtPlugin artPlugin;

    public static ArtPlugin getArtPlugin() {
        return artPlugin;
    }

    @Override
    public void onLoad() {
        artPlugin=this;
        load();
    }

    @Override
    public void onEnable() {
enable();
    }

    @Override
    public void onDisable() {
disable();
   }

    public abstract void load();
    public abstract void enable();
    public abstract void disable();



}

File: ArtCoreKT/src/main/kotlin/ArtPlugin.kt
package net.dev.art.core.artktcore

import org.bukkit.plugin.java.JavaPlugin

abstract class ArtPlugin : JavaPlugin() {

    override fun onLoad() {
        super.onLoad()
    }

    override fun onEnable() {
    }

    override fun onDisable() {
        super.onDisable()
    }

    abstract fun aoHabilitar()
    abstract fun aoCarregar()
    abstract fun aoDesablilitar()

}
